oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/authentication/ClientSecretAuthenticationProvider.java

```diff
@@ -47,6 +50,7 @@
  */
 public final class ClientSecretAuthenticationProvider implements AuthenticationProvider {
 	private static final String ERROR_URI = "https://datatracker.ietf.org/doc/html/rfc6749#section-3.2.1";
+	private static final Log logger = LogFactory.getLog(ClientSecretAuthenticationProvider.class);
```

Comment:
I wonder if we should change this to `private final Log logger = LogFactory.getLog(getClass())` for all `AuthenticationProvider`'s?

Reply:
Sounds good. I couldn't find consensus in Spring Security on it, and different projects tend to prefer different ways (`LOGGER` vs `logger`, static vs instance, etc.)